test/integration/testConnection.js

```diff
@@ -136,6 +138,61 @@ describe('Connection test', function () {
         }
       });
   });
+
+  it('When connect async with original callback then successfully established', async function () {
```

Comment:
Since these tests are so similar, I'd suggest to just pass connection func as an argument to a common method.